enhancements/machine-config/install-time-support-for-image-mode-on-openshift.md

```diff
@@ -0,0 +1,342 @@
+4. The control plane node which is running the MachineOSBuilder pod establishes connection with external image registry
+5. The control plane node which is running the MachineOSBuilder pod pushes the built image to the registry
+6. MCO/MCC creates a MachineConfig to override the `osImageURL` for the worker pool to point to the newly built image
+7. The worker nodes are created with the stock AMI
+8. The MCO runs `bootstrap switch` or `rpm-ostree rebase` to boot the worker nodes into the custom built OS image
+
+#### Notes and Restrictions:
+
+1. The Bootstrap node cannot communicate with the worker nodes
+2. The MOSC and MOSB for the master pool will have to be mocked and applied to the cluster so that when the MachineOSBuilder is started, it knows the history of the image built and applied to the master pool so we don't end up rebuilding unnecessarily
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+The primary use-case for Image Mode in Hypershift is for the host cluster admins to quickly roll out hotfixes and other changes to all of the guest cluster nodes (in addition to control plane nodes). The guest cluster owners do not need to customize their OS. Some adaptations would be required to Hypershift in order to traverse the host <-> guest cluster barrier. It is worth noting that within a Hypershift setup, the management cluster may be several versions behind the guest clusters or may not even be running OpenShift at all. Additionally, the guest clusters can each run different versions of OCP, meaning that it would not be appropriate to build a single base image to be used for all guest cluster nodes. Instead, images should be built for each guest clusters' OS / extensions image.
+
+We currently don't have support for Hypershift with post-install Image Mode. Install time support for this will be added (if needed) once we have post-install support for it. This is currently not part of the implementation plan for this enhancement. **SHOULD Hypershift support be in the Non-Goals section too?**
+
+#### Standalone Clusters
+
+Clusters in disconnected environments require special environment configurations such as a local package mirror, local image registry, etc.
+Post-install Image Mode currently works for disconnected environments and should work in a similar manner for install time
+
+#### SNO or MicroShift
+
+Single-Node OpenShift (SNO) currently works fine with post-install Image Mode and should continue to work as-is for install time support as well. However, there may be some efficiency improvements that can be made. For example, the final image could be written directly to the nodes’ filesystem before the handoff to rpm-ostree / bootc to avoid a registry roundtrip or a minimal registry should be started on the bootstrap node to host the built image. These efficiency gains would be best handled as a separate enhancement.
```

Comment:
SNO still has a bootstrap node as well right? Or are all SNO configurations pivot-in-place from bootstrap to master+worker?